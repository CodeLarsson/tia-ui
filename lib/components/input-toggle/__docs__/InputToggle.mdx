import { Canvas, Meta } from '@storybook/blocks';
import Example from './Example.tsx';
import * as InputToggle from './InputToggle.stories.tsx';

<Meta of={InputToggle} title="InputToggle" />

# InputField

InputField component

#### Example

<Canvas of={InputToggle.PrimaryMedium} />

## Usage

```ts
import {InputToggle} from "tia-ui";

const Example = () => {
  return (
      <InputToggle
        label={"Label"}
        displayLabel={false}
        size={"medium"}
        type={"primary"}
        checked={false}
        onChange={(v) => onChange(v)}
        id={id}
        disabled={disabled}
      />
  );
};

export default Example;
```

#### Arguments

- **label** - `string`, the label of the input element
- **size** - A string with one of three possible values: "small," "medium," or "large." It defines the size or dimensions of the input field.
- **id** - `string`, the id of the DOM element, if none provided, a random generated id will be provided.
- **disabled** - `boolean` indicating whether the field should be disabled or not.
- **checked** - `boolean`, if the toggle is toggled or not
- **displayLabel** - `boolean`, controls if the label should be displayed or not
- **onChange** _`(value: boolean) => void`_
- **type** - `string`, "primary" or "secondary"
