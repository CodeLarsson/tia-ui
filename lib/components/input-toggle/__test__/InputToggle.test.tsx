import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import InputToggle from '../input-toggle';

describe('InputToggle component', () => {
  it('InputToggle should render correctly', () => {
    // Arrange
    const { container } = render(<InputToggle label="Test" />);
    const input = container.querySelector('input');

    // Assert
    expect(input).toBeTruthy();
    expect(input?.value).toBe('false');
  });

  it('InputToggle should render label correctly', () => {
    // Arrange
    const { container } = render(<InputToggle label="Test" />);
    const label = container.querySelector('label');

    // Assert
    expect(label).toBeTruthy();
    expect(label?.textContent).toBe('Test');
  });

  it('InputToggle should render id correctly', () => {
    // Arrange
    const { container } = render(<InputToggle label="Test" id="test" />);
    const input = container.querySelector('input');

    // Assert
    expect(input).toBeTruthy();
    expect(input?.id).toBe('test');
  });

  it('InputToggle should render onChange correctly', () => {
    // Arrange
    const onChange = vi.fn();
    const { container } = render(
      <InputToggle label="Test" onChange={onChange} id="test" />
    );

    const toggle = container.querySelector('#test-toggle-button');
    const input = screen.getByLabelText<HTMLInputElement>('Test');

    // Assert
    expect(toggle).toBeTruthy();
    expect(input).toBeTruthy();
    expect(input?.value).toEqual('false');
    expect(input?.checked).toBeFalsy();

    // Act
    fireEvent.click(toggle!);

    // Assert
    expect(onChange).toHaveBeenCalled();
    expect(input?.value).toEqual('true');
    expect(input?.checked).toBeTruthy();
  });

  it('InputToggle should render disabled correctly', () => {
    // Arrange
    const onChange = vi.fn();
    const { container } = render(
      <InputToggle label="Test" onChange={onChange} id="test" disabled />
    );

    const toggle = container.querySelector('#test-toggle-button');
    const input = screen.getByLabelText<HTMLInputElement>('Test');

    // Assert
    expect(toggle).toBeTruthy();
    expect(input).toBeTruthy();
    expect(input?.value).toEqual('false');
    expect(input?.checked).toBeFalsy();
    expect(input?.disabled).toEqual(true);

    // Act
    fireEvent.click(toggle!);

    // Assert
    expect(onChange).not.toHaveBeenCalled();
    expect(input?.value).toEqual('false');
    expect(input?.checked).toBeFalsy();
  });

  it('Input toggle should match snapshot - primary', () => {
    // Arrange
    const { container } = render(
      <InputToggle label="Test" type="primary" id="test" />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('Input toggle should match snapshot - secondary', () => {
    // Arrange
    const { container } = render(
      <InputToggle label="Test" type="secondary" id="test" />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
