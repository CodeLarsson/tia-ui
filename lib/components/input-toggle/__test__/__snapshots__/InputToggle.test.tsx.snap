// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputToggle component > Input toggle should match snapshot - primary 1`] = `
<div>
  <div
    class="tia-flex tia-flex-col"
  >
    <label
      class="tia-font-semibold tia-mx-2 tia-my-1 tia-text-gray-900 tia-text-md"
      for="test"
    >
      Test
    </label>
    <div
      aria-checked="false"
      class="tia-bg-primary-500 tia-relative tia-flex tia-items-center tia-rounded-full tia-w-medium tia-h-medium "
      role="switch"
    >
      <div
        class="tia-absolute tia-left-[2px] tia-outline-none tia-transform tia-transition-transform tia-duration-200 tia-rounded-full tia-cursor-pointer tia-bg-primary-50 focus:tia-bg-primary-200 tia-w-[2.5rem] tia-h-[2.5rem]  "
        id="test-toggle-button"
        tabindex="0"
      />
      <input
        aria-label="Test"
        class="tia-hidden"
        id="test"
        readonly=""
        type="checkbox"
        value="false"
      />
    </div>
  </div>
</div>
`;

exports[`InputToggle component > Input toggle should match snapshot - secondary 1`] = `
<div>
  <div
    class="tia-flex tia-flex-col"
  >
    <label
      class="tia-font-semibold tia-mx-2 tia-my-1 tia-text-gray-900 tia-text-md"
      for="test"
    >
      Test
    </label>
    <div
      aria-checked="false"
      class="tia-bg-white tia-border tia-border-primary-500 tia-border-solid tia-relative tia-flex tia-items-center tia-rounded-full tia-w-medium tia-h-medium "
      role="switch"
    >
      <div
        class="tia-absolute tia-left-[2px] tia-outline-none tia-transform tia-transition-transform tia-duration-200 tia-rounded-full tia-cursor-pointer tia-bg-primary-500 focus:tia-bg-primary-300 tia-w-[2.5rem] tia-h-[2.5rem]  "
        id="test-toggle-button"
        tabindex="0"
      />
      <input
        aria-label="Test"
        class="tia-hidden"
        id="test"
        readonly=""
        type="checkbox"
        value="false"
      />
    </div>
  </div>
</div>
`;
